eng/scripts/Generate-Patch.ps1

```diff
@@ -244,7 +249,12 @@ if(!$BranchName) {
 
 try {
   ## Creating a new branch
-  $cmdOutput = git checkout -b $BranchName $RemoteName/main
+  if($CreateNewBranch) {
```

Comment:
Depending on the branch name you might hit issues if the branch already exists.

Reply:
Most likely you will hit those issues one push. You might need to harden the logic for retries, or potentially use our common script for that which already handles that.

Reply:
Do you mean challenges when we are pushing the changes to the upstream? Would we hit those even if we pushed using '-force' flag? Can you point me to the script?

My thought here was 
(a) Most people would create a new branch for this.
(b) Some people may want to override by using special scenarios like storage - they use a single release tag to release all their SDK libraries. These people know what the expectations are from the branch, so will manually hold things through.

Reply:
The script I was referring to is at https://github.com/Azure/azure-sdk-for-java/blob/main/eng/common/scripts/git-branch-push.ps1. 

We want to be careful about force pushing to a branch, especially in the main repo as someone might accidently give a branch with the same name as something else which could cause work to get lost. If we do force push I suggest we push to a fork and not the main branch (perhaps you already are but if not you should consider that).

Reply:
Got it. I actually try to push to the main repo since pipelines are rooted to it. Thanks for pointing out. I will remove the force push and use the common script instead.

Reply:
I looked at the script and I do not think I should use it in this scenario. The common script assumes that multiple teams in theory could be working on the same branch, while this script assumes it is in isolation per service tree.  I do think I should remove the force push. However I will err toward caution and let the script fail to push the changes if the remote branch exists.

Reply:
The common script is definitely more hardened to error cases and I do think it would work for your scenarios, but I'm also fine with you doing things explicitly here. If in when you need to make more updates to fix other scenarios you should consider using the common script.

Reply:
For example the common script will check to see if the branch exists and call the correct checkout command for either condition without needing to pass in a boolean.

Reply:
Yes. I think this script will be really helpful when we move to automating patch releases through pipeline but I don't think we are there yet. I want to spend a few more iterations before attempting that :)